media-gfx/openvdb/openvdb-11.0.0.ebuild

```diff
@@ -0,0 +1,361 @@
+		media-libs/libglvnd
+	)
+	!ax? (
+		dev-libs/log4cplus:=
+	)
+"
+
+DEPEND="${RDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+	doc? (
+		app-text/doxygen
+		dev-texlive/texlive-bibtexextra
+		dev-texlive/texlive-fontsextra
+		dev-texlive/texlive-fontutils
+		dev-texlive/texlive-latex
+		dev-texlive/texlive-latexextra
+	)
+	test? (
+		dev-cpp/gtest
+		dev-util/cppunit
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-8.1.0-glfw-libdir.patch"
+
+	"${FILESDIR}/${PN}-9.0.0-fix-atomic.patch"
+
+	"${FILESDIR}/${PN}-10.0.1-fix-linking-of-vdb_tool-with-OpenEXR.patch"
+	"${FILESDIR}/${PN}-10.0.1-log4cplus-version.patch"
+
+	"${FILESDIR}/${PN}-11.0.0-constexpr-version.patch"
+	"${FILESDIR}/${PN}-11.0.0-cmake_fixes.patch"
+)
+
+cuda_set_CUDAHOSTCXX() {
+	local compiler
+	tc-is-gcc && compiler="gcc"
+	tc-is-clang && compiler="clang"
+	[[ -z "$compiler" ]] && die "no compiler specified"
+
+	local package="sys-devel/${compiler}"
+	local version="${package}"
+	local CUDAHOSTCXX_test
+	while
+		CUDAHOSTCXX="${CUDAHOSTCXX_test}"
+		version=$(best_version "${version}")
+		if [[ -z "${version}" ]]; then
+			if [[ -z "${CUDAHOSTCXX}" ]]; then
+				die "could not find supported version of ${package}"
+			fi
+			break
+		fi
+		CUDAHOSTCXX_test="$(
+			dirname "$(
+				realpath "$(
+					which "${compiler}-$(echo "${version}" | grep -oP "(?<=${package}-)[0-9]*")"
+				)"
+			)"
+		)"
+		version="<${version}"
+	do ! echo "int main(){}" | nvcc "-ccbin ${CUDAHOSTCXX_test}" - -x cu &>/dev/null; done
+
+	export CUDAHOSTCXX
+}
+
+cuda_get_host_arch() {
+	[[ -z "${CUDAARCHS}" ]] && einfo "trying to determine host CUDAARCHS"
+	: "${CUDAARCHS:=$(__nvcc_device_query)}"
+	einfo "building for CUDAARCHS = ${CUDAARCHS}"
+
+	export CUDAARCHS
+}
+
+pkg_setup() {
+	if use nanovdb && use cuda; then
+		einfo "OpenVDB must be compiled with TBB. TBB fails when compiled for CUDA. Using NANOVDB_USE_OPENVDB=OFF."
+	fi
+
+	use ax && llvm_pkg_setup
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	MULTIBUILD_VARIANTS=( install )
+	use test && MULTIBUILD_VARIANTS+=( test )
+
+	rm "cmake/Find"{OpenEXR,TBB}".cmake" || die
+
+	if use nanovdb; then
+		sed \
+			-e 's#message(WARNING " - OpenVDB required to build#message(VERBOSE " - OpenVDB required to build#g' \
+			-i "nanovdb/nanovdb/"*"/CMakeLists.txt" || die
+	fi
+
+	cmake_src_prepare
+
+	sed -e 's|/usr/local/bin/python|/usr/bin/python|' \
+		-i "${S}"/openvdb/openvdb/python/test/TestOpenVDB.py || die
+}
+
+my_src_configure() {
+	local version
+	version=$(ver_cut 1)
+	if use "abi$(( version + 1 ))-compat"; then
+		version=$(( version + 1 ))
+	elif use "abi$(( version - 1 ))-compat"; then
+		version=$(( version - 1 ))
+	elif use "abi$(( version - 2 ))-compat"; then
+		version=$(( version - 2 ))
+	fi
+
+	local mycmakeargs=(
+		-DCMAKE_FIND_PACKAGE_PREFER_CONFIG="yes"
+		-DCMAKE_INSTALL_DOCDIR="share/doc/${PF}/"
+
+		-DOPENVDB_ABI_VERSION_NUMBER="${version}"
+		-DOPENVDB_BUILD_DOCS="$(usex doc)"
+		-DOPENVDB_BUILD_UNITTESTS="$(usex test)"
```

Comment:
This is a bit annoying, but you may want to group these under `if use test ; then` block:
```
 * One or more CMake variables were not used by the project:
 *   BUILD_TEST
 *   NANOVDB_BUILD_UNITTESTS
 *   OPENVDB_BUILD_PYTHON_UNITTESTS
```